ansible_base/lib/logging/filters/request_audit_info.py

```diff
@@ -0,0 +1,39 @@
+import logging
+
+from ansible_base.lib.logging import thread_local
+
+
+class RequestAuditInfoFilter(logging.Filter):
+    def filter(self, record):
+        """
+        This "filter" is used to add request information to the log record.
+        It extracts the source IP and user agent from the request.
+        It will always return True, so that the message is always logged,
+        even if there is no request.
+        """
+
+        # Always set defaults so we can use them in the logging formatter.
+        record.source_ip = ""
+        record.user_agent = ""
+
+        request = getattr(thread_local, "request", None)
```

Comment:
I suppose this general structure could be abstractified into some kind of nonblocking request logging filter with plugins to reduce the overlap with RequestIdFilter, but no big deal there.  Maybe if a 3rd one comes along..